src/server/zset_family.cc

```diff
@@ -2030,6 +2030,52 @@ void ZSetFamily::ZInterStore(CmdArgList args, ConnectionContext* cntx) {
   (*cntx)->SendLong(smvec.size());
 }
 
+void ZSetFamily::ZInter(CmdArgList args, ConnectionContext* cntx) {
+  OpResult<SetOpArgs> op_args_res = ParseSetOpArgs(args, false);
+
+  if (!op_args_res) {
+    return HandleOpStatus(cntx, op_args_res);
+  }
+  const auto& op_args = *op_args_res;
+  if (op_args.num_keys == 0) {
+    return SendAtLeastOneKeyError(cntx);
+  }
+
+  vector<OpResult<ScoredMap>> maps(shard_set->size(), OpStatus::SKIPPED);
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    maps[shard->shard_id()] = OpInter(shard, t, "", op_args.agg_type, op_args.weights, false);
+    return maps[shard->shard_id()];
+  };
+
+  cntx->transaction->ScheduleSingleHopT(std::move(cb));
+
+  ScoredMap result;
+  for (auto& op_res : maps) {
+    if (op_res.status() == OpStatus::SKIPPED)
+      continue;
+
+    if (!op_res)
+      return (*cntx)->SendError(op_res.status());
+
+    if (op_res->empty())
+      break;
+
+    if (result.empty()) {
+      result.swap(op_res.value());
+    } else {
+      InterScoredMap(&result, &op_res.value(), op_args.agg_type);
+    }
+  }
+
+  std::vector<std::pair<std::string, double>> scoredArray;
+  for (const auto& elem : result) {
+    scoredArray.emplace_back(elem.first, elem.second);
+  }
+
+  (*cntx)->SendScoredArray(scoredArray, op_args_res->with_scores);
```

Comment:
I just read Redis' documentation. It's true that they do not explicitly mention sorting, but they say that this is like `ZINTERSTORE` which stores the result in a sorted set. And sorted sets are ordered.
Further, and more importantly, I just tested this on Redis, and no matter which order I insert elements and intersect them, I always get the results ordered. So I think we *should* be ordering, to be compatible with Redis.
It shouldn't be hard, though, something like:

```cpp
std::sort(scored_array.begin(), scored_array.end(), [](const d::pair<std::string, double>& a, const d::pair<std::string, double>& b) { return a.second < b.second; });
```

Reply:
Thanks for your investigate!
fixed: https://github.com/dragonflydb/dragonfly/pull/2164/commits/b5516ac9ff0eef0196824907d33adc3d2b701426